src/js/utils/MesosStateUtil.js

```diff
@@ -87,6 +92,98 @@ const MesosStateUtil = {
   },
 
   /**
+   * Return historical instances (killed, terminated, failed etc.) for the given
+   * pod by digging into the marathon state.
+   *
+   * @param {Object} state - The mesos state response
+   * @param {Pod} pod - The related pod
+   * @returns {Array} The array of historical instances
+   */
+  getPodHistoricalInstances(state, pod) {
+    let frameworks = state.frameworks || [];
+    let marathonFramework = frameworks.find(function (framework) {
+      return framework.name === 'marathon';
+    });
+
+    if (!marathonFramework) {
+      return [];
+    }
+
+    let instancesMap = marathonFramework.completed_tasks.reduce(
+      function (memo, task) {
+        if (POD_TASK_REGEX.test(task.id)) {
+          let [, podID, instanceID] = POD_TASK_REGEX.exec(task.id);
+          if (podID === pod.getMesosId()) {
+            let containerArray = memo[instanceID];
+            if (containerArray === undefined) {
+              containerArray = memo[instanceID] = [];
+            }
+
+            // The last status can give us information about the time the
+            // container was last updated, so we need the latest status item
+            let lastStatus = task.statuses.reduce(function (memo, status) {
+              if (!memo || (status.timestamp > memo.timestamp)) {
+                return status;
+              }
+              return memo;
+            }, null);
+
+            // Add additional fields to the task structure in order to make it
+            // as close as possible to something a PodContainer will understand.
+            containerArray.push(Object.assign({
+              containerId: task.id,
+              status: task.state,
+              //
+              // NOTE: We are creating a Date object from this value, so we
+              //       should be OK with the timestamp
+              //
+              lastChanged: lastStatus.timestamp * 1000,
+              lastUpdated: lastStatus.timestamp * 1000
+            }, task));
+          }
+        }
+
+        return memo;
+      }, {}
+    );
+
+    // Try to compose actual PodInstance structures from the information we
+    // have so far. Obviously we don't have any details, but we can populate
+    // most of the UI-interesting fields by summarising container details
+    return Object.keys(instancesMap).map(function (instanceId) {
+      let containers = instancesMap[instanceId];
+      let summaryProperties = containers.reduce(function (memo, instance) {
+        let {resources={}, lastChanged=0} = instance;
+
+        memo.resources.cpus += resources.cpus || 0;
```

Comment:
I think this would be cleaner if we are assigning values, but in this case we are adding. From what I see this would make things more complicating for 4 lines of code. Can you write down your suggestion?